tests/incident/test_views.py

```diff
@@ -178,3 +178,103 @@ def test_incident_search_incident_and_event(self):
         self.add_event(pk2, "target_event")
         response = self.client.get("/api/v2/incidents/?search=target")
         self.assertEqual(len(response.data["results"]), 2)
+
+
+class BulkAcknowledgementViewSetTestCase(APITestCase):
+    @classmethod
+    def setUpClass(cls):
+        disconnect_signals()
+        source_type = SourceSystemTypeFactory()
+        cls.user = SourceUserFactory()
+        cls.source = SourceSystemFactory(type=source_type, user=cls.user)
+
+    def setUp(self):
+        self.client.force_authenticate(user=self.user)
+
+    @classmethod
+    def tearDownClass(cls):
+        connect_signals()
+
+    def add_incident(self, description="incident"):
+        data = {
+            "start_time": "2021-08-04T09:13:55.908Z",
+            "end_time": "2021-08-04T09:13:55.908Z",
+            "description": description,
+            "level": 1,
+            "tags": [{"tag": "a=b"}],
+        }
+        response = self.client.post("/api/v2/incidents/", data, format="json")
+        self.assertEqual(response.status_code, 201)  # Created
+        return response.data["pk"]
+
+    def test_can_bulk_create_acknowledgements_for_incidents_with_valid_ids(self):
+        incident_1_pk = self.add_incident()
+        incident_2_pk = self.add_incident()
+        self.assertFalse(Acknowledgement.objects.exists())
+        data = {
+            "ids": [incident_1_pk, incident_2_pk],
+            "ack": {
+                "event": {
+                    "timestamp": "2022-08-02T13:04:03.529Z",
+                    "description": "acknowledgement",
+                },
+                "expiration": "2022-08-03T13:04:03.529Z",
+            },
+        }
+        response = self.client.post(path=f"/api/v2/incidents/acks/bulk/", data=data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(response.data["changes"][str(incident_1_pk)]["status"], status.HTTP_201_CREATED)
+        self.assertEqual(response.data["changes"][str(incident_1_pk)]["ack"]["event"]["type"]["value"], "ACK")
+        self.assertEqual(response.data["changes"][str(incident_1_pk)]["errors"], None)
+        self.assertEqual(response.data["changes"][str(incident_2_pk)]["status"], status.HTTP_201_CREATED)
+        self.assertEqual(response.data["changes"][str(incident_2_pk)]["ack"]["event"]["type"]["value"], "ACK")
+        self.assertEqual(response.data["changes"][str(incident_2_pk)]["errors"], None)
```

Comment:
I find these assertions hard to read. Maybe extract the changes response dictionary for each PK individually and use those in the asserts?

```suggestion
        incident_1_changes = response.data["changes"][str(incident_1_pk)]
        self.assertEqual(incident_1_changes["status"], status.HTTP_201_CREATED)
        self.assertEqual(incident_1_changes["ack"]["event"]["type"]["value"], "ACK")
        self.assertEqual(incident_1_changes["errors"], None)
        
        incident_2_changes = response.data["changes"][str(incident_2_pk)]
        self.assertEqual(incident_2_changes["status"], status.HTTP_201_CREATED)
        self.assertEqual(incident_2_changes["ack"]["event"]["type"]["value"], "ACK")
        self.assertEqual(incident_2_changes["errors"], None)
```